WordPress/gutenberg
# Introduce the `customScripts` property to allow templates to define additional scripts in package.json

## Description
As referenced in https://github.com/WordPress/gutenberg/issues/25188#issuecomment-1030044917, this PR adds a `customScripts` property to templates to add additional custom scripts to the package.json. One use-case would be to add e2e tests ie. `"test:e2e": "wp-scripts test-e2e"`.

## Testing Instructions
* `./node_modules/.bin/wp-create-block` to ensure nothing is added
* Download https://github.com/ryanwelcher/dynamic-block-template and add a `customScripts` entry to the `index.js`
* Run `./node_modules/.bin/wp-create-block --template={pathTo}/dynamic-block-template`

## Types of changes
New Feature

## Checklist:
